[Pattern Draft] Circle Communities (#696)
gruetter/innerSourceTest: patterns/1-initial/circle-communities.md

* **Knowledge Sharing and Best Practices**: Discussions include reviewing best practices such as how to promote openness in code review, establish transparent governance, or mentor contributors effectively.

To nurture Circle Communities, implement these practices:

* **Define the Purpose**: Clearly articulate goals and align them with the organization's mission.
* **Establish Social Conventions**: Develop guidelines to promote respect, inclusivity, and open dialogue, evolving them over time.
* **Facilitate Regular Meetings**: Schedule consistent meetings for progress discussions, insight sharing, and collaborative problem-solving.
* **Rotate Leadership**: Encourage shared leadership to distribute responsibility and deepen member connections.
* **Leverage Collaborative Tools**: Use tools like Slack or GitHub to facilitate communication, documentation, and privacy-preserving insights sharing.
* **Foster Continuous Improvement**: Solicit regular feedback to refine the community's activities and structure.

> ## Resulting Context

Author: @spier - I refactored the Resulting Context section as requested.

Please advise.

## Title

Circle Communities

## Patlet

> InnerSource adoption is slow in organizations due to limited understanding, engagement, and contextual relevance. Circle Communities address this by fostering synchronous conversations that build connections, close knowledge gaps, and cultivate collaboration and continuous learning.

Author: @spier - I refactored the Patlet section as requested.

Please advise.

## Title

Circle Communities

## Patlet

> InnerSource adoption in organizations is slow due to limited understanding and engagement. Circle Communities foster a culture of collaboration and continuous learning, which are essential to InnerSource. These communities provide a platform for synchronous conversations, where knowledgeable and confident members can engage in active listening and help build the necessary connections and understanding.

Reviewer: Can we try to get this down to two sentences?
1st sentence / the problem
2nd sentence / the solution and how it contributes to solving the problem

it is hard but it is also a good exercise as it forces us to answer the question specifically:
How does this solution help us to improve the given InnerSource challenge?

Author: I placed a fresh comment on the file with the refactored language.

Please advise (there ideally).

## Problem

Organizations face several challenges in building individual understanding around InnerSource and growing a stakeholder community. Problems can be broken down into a few sub-categories:

* **Lack of Contextual Understanding**: Employees often struggle with relevance about InnerSource, hindering their learning process.
* **Knowledge Gaps**: There is often a significant gap between what individuals know and what they need to learn, making it difficult to engage meaningfully.
* **Engagement Issues**: Without predictable and structured opportunities for synchronous conversations, engagement and participation can be low.
* **Limited Expertise**: Organizations may lack sufficient confident and knowledgeable individuals who can facilitate discussions and bridge the knowledge gaps.

## Context

> Scenario 1: Few InnerSource champions exist within your heavily siloed organization. The practice has struggled to build sustainable engagement as the culture is not hospitable to open collaboration and innovation.

Reviewer: ```suggestion
Scenario 1: Few people with extensive InnerSource knowledge exist within your heavily siloed organization. The practice has struggled to build sustainable engagement as the culture is not hospitable to open collaboration and innovation.
```

I would not use the word "champion" here, as we are trying to establish ambassador/champion as a pattern and term in its own right.

Therefore I would rather describe what is meant here. I tried but not sure if I got it right?

Scenario 2: Your organization is decentralized and often stakeholders are unaware of other projects or programs. InnerSource has struggled to get engagement as the culture is highly dispersed and segmented across the organization.

> ## Forces

Reviewer: The Forces section should describe what makes the problem difficult, and should not contain hints to the solution yet.

The 4 bullets from the Problem section ("Lack of Contextual Understanding" etc) could be used as Forces.

The bullets that are currently listed as forces ("Diversity of Participants" etc) could be moved to the Solution instead.

Author: I have taken another pass at the Problem, Forces and Solutions sections based on your feedback.

I also considered with fresh eyes some distinctions from other types of Communites of Practice.

Please advise @spier

Reviewer: These changes work for me. Nice job!
Let's leave these as they are.

* **Diversity of Participants**: Circle Communities often bring together individuals with different backgrounds, expertise and expectations.
* **Need for Safe Spaces**: Members must feel safe to share their thoughts, experiences and perspectives.
* **Continuous Engagement**: Sustained participation is crucial for the longevity and impact of the community.
* **Knowledge Sharing**: Effective mechanisms for sharing knowledge and best practices are needed.

> ## Solutions

Reviewer: Is it possible to be more specific about what a Circle Community around InnerSource would look like?

Currently the description is more a general "this is how to start a Circle Community/Guild/Community of Practice" and contains no specifics related to InnerSource.

Author: @spier - Very fair point ... I reflected on this and restated the key sections to clarify the distinction.

Please advise.

Reviewer: Yes, it is certainly better now. Let's leave this as is.

Circle Communities are a collaboration forum based community. Participation is voluntary. Except for the duties of a facilitator, no distinction is enforced between participants. This invokes round-table equality which maximizes authenticity and diversity. Circle Communities often bring together individuals with different backgrounds, expertise and expectations.  Members must feel safe to share their thoughts, experiences and perspectives.  Sustained participation is crucial for the longevity and impact of the community.  Effective mechanisms for sharing knowledge and best practices are needed.

Implement a strategic approach to forming and nurturing Circle Communities that emphasizes clear objectives, inclusive participation and continuous improvement.  This can be achieved through the following parallel efforts:

> * **Define the Purpose**: Clearly articulate the purpose and goals of the Circle Community.  Ensure that these align with the broader mission of your organization or initiative.

Reviewer: Here I realize that I am not sure what this pattern is about, at its core.

Two theories:

Is this pattern about starting a **single** Circle Community around InnerSource?
And then that circle/guild will specifically try to increase InnerSource adoption in the org?

Or is the pattern about starting **multiple** Circle Communities (about various topics)?
And then practicing these circles/guilds will eventually lead to a certain mindset shift and change in collaboration approach, which is the basis for making InnerSource possible?

Author: I addressed this in another comment near the end of the file with some tangentially connected (in my mind) questions.

Please resolve or let me know if you have more questions (here or there no difference to me).

* **Foster a Culture of Continuous Improvement**: Regularly solicit feedback from members to identify areas for improvement. Use this feedback to refine the conditioning and activities of the Circle Community.

## Resulting Context

Embracing this pattern helps InnerSource programs become vibrant, effective and sustainable. Participants will feel more connected and engaged, leading to increased knowledge sharing, innovation and collective growth.

## Related Patterns

* [Dedicated Community Leader](../2-structured/dedicated-community-leader.md)
* [Core Team](../2-structured/core-team.md)
* [Maturity Model](../2-structured/maturity-model.md)
* [Agile InnerSource Dojo](../1-initial/agile-innersource-dojo.md)

## Known Instances

> * Dojo Circle

Reviewer: Is the Dojo Circle an organization that applies this pattern to solve a problem that they faced while implementing InnerSource?

Author: I believe this has been removed from another commit.

If you agree please resolve.  

If not, please help me understand where this shows up in the refreshed version.

> Embracing this pattern helps InnerSource programs become vibrant, effective and sustainable. Participants will feel more connected and engaged, leading to increased knowledge sharing, innovation and collective growth.

Reviewer: I cannot clearly see the connection between the Solution and the Resulting Context.

Can we describe more specifically how the Solution is improving upon the listed Problem?

Author: I placed a fresh comment on the file with the refactored language.

Please advise (there ideally).

> 

Reviewer: ```suggestion
```

Moving this section further down.

## Known Instances

* [InnerSource Dojo from SAP](https://www.youtube.com/watch?v=fXoVm5iTSCc)
* _Unplugged InnerSource_ sessions within the _Reuse & InnerSource_ workgroup of [SAP Cross-Platform Architecture](https://community.sap.com/t5/technology-blogs-by-sap/cross-product-architecture-embracing-conway-s-law-for-better-software/ba-p/13648600)
* [Guilds at the BBC](https://youtu.be/MyYgk-jdjSw?t=511)

> * [Mindset Dojo Community](https://dojo.center)

Reviewer: ```suggestion
```

Known Instances refer to organizations that have implemented this pattern.

This looks like it is a commercial offering from you where you want to teach people about this and other concepts? So in our definition this would not be a Known Instance and should be removed.

Author: Mindset Dojo Community is pivoting a bit.  Still in early infancy.   This is a bit of a more expansive conversation that I don't want to cloud this pattern itself.

Arguably the practitioners within Mindset Dojo may bring this pattern into organizations they work with and then those instances would have article / case study linkages directly.

In my new role starting on the 13th I am eager to consider bringing this pattern in when the timing and forces are in alignment.   

For example, based on what I already know there may be potential interest, but I don't know for certain.  

As such, I have removed as you requested.

If you agree this is now done, please resolve.

> 

Reviewer: ```suggestion

## Related Patterns

* [Dedicated Community Leader](../2-structured/dedicated-community-leader.md)
* [Core Team](../2-structured/core-team.md)
* [Maturity Model](../2-structured/maturity-model.md)

```

## Status

* Initial

## Authors

* [Michael Basil](https://www.linkedin.com/in/michaelrbasil)
* [Addie Girouard](https://www.linkedin.com/in/addiegirouard)
* [Bill Westfall](https://www.linkedin.com/in/bill-westfall-3268494)
* [Jim Manzzullo](https://www.linkedin.com/in/jimmanzzullo)
* [Jeremy McMillan](https://www.linkedin.com/in/jeremymcm)
* [AnuOlu Balogun](https://www.linkedin.com/in/anuolu)

> ## Acknowledgments

Reviewer: These are a lot of acknowledgments :)
What is your goal with acknowledging all of them?
Did they all indirectly contribute to this pattern?

I am commenting on this as the combined list of authors/acknowledgements is easily the longest of all of our existing patterns. And while some (self-)promotion and name dropping is certainly fine, I would not want this to become too extensive.

Author: Each person in this list participated to help cultivate this pattern.   I would remind that Bill, AnuOlu, Jim and I presented on the core concept underlying this pattern to the InnerSource Commons in early-mid 2024.  Jeremy was instrumental just as much or more than other names on that list and arguably could be moved up a slot or two.

This is a highly socially collaborative pattern and the lists reflects that. 

I feel all of the author listings and acknowledgements are warranted in such a socially-engaged pattern.

There was no rule stated coming in about the length of the list and it appears this is just a divergence from the current expected norm.

If you feel strongly that you want to set limits here, I would state my objection as that seems to be against the spirit of spreading the patterns themselves and then ask for direct clarity on the limits and then I would make a modification to comply to move the pattern forward.

Please let me know on how to proceed and or if this comment can be resolved.

Reviewer: If you want to keep the whole list as is, that is fine with me as well.

Your comment above seems to refer mostly to the people in the Authors list. Does the same reasoning apply to the Acknowledgments list?

Author: I ensured that the authors were really involved in the creation of the language of the pattern.  This included conversations and they added their name by a commit in effect "signing" the pattern.   I woudn't sign someone else's name as an author.  The same was true for Addie recently.

Since I am leading the charge to carry this pattern through I did set the order of list based on how influential and supportive people were in this process.

For the acknowledgements list, I have had a conversation with each person though the level to which it would stick would vary.   

I believe the list is the way it should be when I review it now.  That said, if a desire and request was made to pare it down I could do it.  I prefer not to, but I feel far less strongly about it compared to the authors in comparison.

Please let me know how to proceed when you have a moment and or resolve if you feel the way this has been approached meets the spirit and or intention of the section.🙏

Reviewer: Let's leave all of them in then. :)

I guess I was also a bit confused because I had never heard about many of these people in the context of the ISC. Typically I have heard of people one way or the other, before they contribute to a pattern.

However I am just taking this as a way that the ISC is growing in reach, and that you found a way to bring some new people in. So that is amazing :)

Let's leave this as is ❤️

* [Guilherme Dellagustin](https://www.linkedin.com/in/dellagustin)
* InnerSource Dojo from SAP:
  * [Ryan Uttech](https://www.linkedin.com/in/ryanuttech)
  * [Ignacio Pedragon](https://www.linkedin.com/in/ignaciopedregon)
  * [Dale Sackrider](https://www.linkedin.com/in/sackrider)
  * [Jason Fisher](https://www.linkedin.com/in/jasonrfisher)
  * [Michael Ward](https://www.linkedin.com/in/michaelwardqa)
  * [Alfred Darby III](https://www.linkedin.com/in/alfred-darby-iii-21b6683)
* [Nick Basil](https://www.linkedin.com/in/nick-basil)
* [Allison Kaufmann](https://www.linkedin.com/in/allisonkaufmann29)
* [Dojo Consortium](https://dojoconsortium.org)

> * [Thingy Farm](https://thingyfarm.com/)

Reviewer: In this section we typically acknowledge individuals for their indirect contributions to the pattern, not "organizations".

Not sure about Thingy Farm. Could not tell from the website what it is.

Also the Dojo Consortium does not fit out typical approach here.

Author: I'm going to remove these.  They are too much of a stretch for this it seems.  I also feel based on your comment that it would be more likely that Dojo Consortium didn't directly contribute as an organization and is more loosely collected set of DevOps / Agile learning enthusist niche professionals.  There may be interested parties to this pattern to support the introduction of InnerSource as a practice they introduce to organizations in time.  I'm happy to discuss this at some future point if someone approached me on it.

Author: I have removed these.

Please resolve if you agree this is done.

> ## Alias

Reviewer: Which term for this concept is most frequently used in the industry?

Just a single personal data point:
I have heard of guilds and community of practice frequently. I had never heard the term circle communities before.

Reviewer: Also would Dojo be yet another alias for this?
Or is a Dojo more focused on the collaborative practicing of things?

Author: 1. "the industry" is pretty subjective ...

If we are talking broadly speaking then if I am taking "the industry" to include communication and change experts when I talk about a "Circle" community they get an idea of where this is going.

I would consider Circle Communities as a subset of Communities of Practice.  I have ammended several of the sections per other feedback which should suggest where the distinctions are to merit such a separate designation.

I also have conversations with my new employer about a Guild they have (for Agile) that lacks some of the form and flow of a Circle and they are excited to hear more.   I would say that a Circle Community itself is inherently aligned with InnerSource Principles wherase an Agile Circle for example inherets the principles of InnerSource into it's DNA on inception unknowingly.

Moreover, Circle Communties support InnerSource adoption from the bottom up and middle out aligned to top-down or central initiatives whenever they exist and those groups are supportive and willing to engage (always invited).  Support it in specific ways not common to Communities of Practice in general.

2. I would not call a Circle an alias for Dojo.  

The Dojo metaphor and model carries much different distinctions and social-learning system advanced support.  Not all Dojos have Circles, though all Circles could theoretically (if the group desired) spawn and connect to a Dojo learning approach.  I would say though that a Circle could spawn different models as well.  The SAP Dojo featured Circles whereas the Microsoft DevOps Dojo likely did not (though I cannot be sure as I never met anyone directly part of that unfortuanately, at least not yet).

I decided to close the PR we had for a Dojo pattern to allow the focus to center here on the far more important pattern and to consider revisiting if a Dojo pattern made sense, though there would need to be interest and appetite from the InnerSource Commons Community for this (AND IDEALLY participation from Microsoft DevOps Dojo key person(s)). 

The Circle pattern does have affinity and interest from at least one important member who is also listed as an author (you will know the name(s) when you encounter them).  Had that not existed I would have closed the pattern.

I would also add that this Circle Communities pattern is ideal for socially oriented DevOps groups and Agilists.   Additionally, I would like to have a conversation with Cloud Security Office Hours sometime this year after it is published.  Bill and I have presented there in the past and I'm sure he'd be interested and I would love it if someone or multiple people who are with ISC as formal members would participate - but that is not the key point here (just planting an invitational seed for later).  I mention this because I believe outreach into other communities is a conversation I have heard Russ talk about a few times.

3. As far as if this pattern is a single or multiple instance?  

Yes.  More like WHEN and DEPENDING ON what.

Consider it a pattern of flow versus control.   

I have seen it where Circles spawn short and medium lived other Circles or long lived Circle-style collaboration zones in different areas of a business unit after experiencing it.

I say that it starts and ends with one circle.  This is from one point of view though.  Just like a repository codebase there may be multiple "centrals" depending on branching, forking, etc.  Same concept here.

Apologies for the length of this I felt it may be of use to give an expansive answer.  I hope you'll forgive me if it is taxing!  

🙏🌿

Reviewer: There is a lot to unpack here :) I will leave this comment open for future reference.
